### PR TITLE
Prm 286

### PR DESCRIPTION
**Pull request by AndrewRButcher**

### CODE REVIEW

File: app/models/appeals/AppealSubmission.scala [hunk 1]
```diff
@@ -67,20 +77,25 @@ object AppealSubmission {
...
           reasonableExcuse = reasonableExcuse,
           honestyDeclaration = userRequest.session.get(SessionKeys.hasConfirmedDeclaration).get == "true",
           appealInformation = CrimeAppealInformation(
             `type` = "crime",
             dateOfEvent = userRequest.session.get(SessionKeys.dateOfCrime).get,
             reportedIssue = userRequest.session.get(SessionKeys.hasCrimeBeenReportedToPolice).get == "yes",
-            statement = None //TODO: may need to add this in later
+            statement = None,
+            lateAppeal = isLateAppeal,
+            lateAppealReason = userRequest.session.get(SessionKeys.lateAppealReason).getOrElse("") match {
```
Comment (LewisMitchell) [inline app/models/appeals/AppealSubmission.scala:95]: Should be able to replace `userRequest.session.get(SessionKeys.lateAppealReason).getOrElse("")` with `userRequest.session.get(SessionKeys.lateAppealReason)` 

Which would return None if the page hasn't been submitted, or Some("xxx") where the appeal reason has been entered (Some("") if nothing entered but page submitted)?

File: app/services/AppealService.scala [hunk 2]
```diff
@@ -68,7 +72,12 @@ class AppealService @Inject()(penaltiesConnector: PenaltiesConnector) {
   }
 
   def submitAppeal(reasonableExcuse: String)(implicit userRequest: UserRequest[_], ec: ExecutionContext, hc: HeaderCarrier): Future[Boolean] = {
-    val modelFromRequest: AppealSubmission = AppealSubmission.constructModelBasedOnReasonableExcuse(reasonableExcuse)
+    val dateSentParsed: LocalDateTime = LocalDateTime.parse(userRequest.session.get(SessionKeys.dateCommunicationSent).get)
+    val daysResultingInLateAppeal: Int = appConfig.daysRequiredForLateAppeal
+    val dateTimeNow: LocalDateTime = dateTimeHelper.dateTimeNow
+    val isLateApeal = dateSentParsed.isBefore(dateTimeNow.minusDays(daysResultingInLateAppeal))
+
+    val modelFromRequest: AppealSubmission = AppealSubmission.constructModelBasedOnReasonableExcuse(reasonableExcuse, isLateApeal)
```
Comment (LewisMitchell) [inline app/services/AppealService.scala:80]: Typo in `isLateApeal`

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good. AT's seem OK. Just a couple of changes.

File: app/models/appeals/AppealSubmission.scala (hunk 1 above)
Comment (AndrewRButcher) [reply app/models/appeals/AppealSubmission.scala:95]: I tried this initially however an empty string is returned from the session (when the user is presented the question but does not answer) therefore in Wireshark I could see the reason was being sent through to the backend with a null value

Comment (LewisMitchell) [reply app/models/appeals/AppealSubmission.scala:95]: That makes sense, I thought there would be a reason.

File: app/services/AppealService.scala (hunk 2 above)
Comment (AndrewRButcher) [reply app/services/AppealService.scala]: Fixed

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**